HIP/hip-410.md

```diff
@@ -154,28 +170,43 @@ message TransactionBody {
     } 
     //... 
 }
+
+// crypto_get_info.proto
+message CryptoGetInfoResponse {
+    //...
+    message AccountInfo {
+        //...
+        int64 ethereumNonce = 24;
+    }
+    //...
+}
 ```
 
 #### ContractCreateTransaction
 
-In cases where the created contract is small it may fit in a Hedera Transaction.
-Ordinarily contract creates always require the creation of a Hedera file. To
-simplify this case a new field `initcodeBytes` is added to
-the `ContractCreateTransactionBody` as an alternative to `fileID`.
+In the network today, contracts are first uploaded as files in the Hedera File
+Service and then linked with the Smart Contract in a `ContractCreateTransaction`.
+In the Ethereum ecosystem, the code for a smart contract is included in the
+transaction itself.
+
+In cases where the smart contract code is small, it may fit in a Hedera Transaction.
+A new field, `initcodeBytes` will be added to `ContractCreateTransactionBody`. Instead
+of uploading a file and then supplying a `fileID`, you can instead supply the contract
+code directly in the `ContractCreateTransaction`.
 
 ```protobuf
 message ContractCreateTransactionBody {
     oneof initcodeSource {
```

Comment:
We had similar use-cases where a given property had to be changed to `one of` and the discussions that we had with the Mirror node team went in the direction that it is easier (in terms of compatibility) to add another property and implicitly implement the `one of` feature instead of directly changing the protobuf to `one of`.